AtTools/AtEDistortionModel.h

```diff
@@ -0,0 +1,90 @@
+#ifndef ATEDISTORTIONMODEL_H
+#define ATEDISTORTIONMODEL_H
+
+#include "AtSpaceChargeModel.h"
+
+#include <Rtypes.h> // for Double_t
+
+#include <fstream>
+#include <type_traits> // for add_pointer_t
+#include <unordered_map>
+
+class AtDigiPar;
+/**
+ * @brief Model for E-field distortion correction
+ *
+ *
+ *
+ *     The implemented correction is applied to every hit
+ *     Radial, transversal and longitudinal (z) corrections as a function of radial distance and z (map keys).
+ *     The Class EField mapping contains the correction values
+ */
+
+struct EFieldCorr {
+   EFieldCorr() : _zcorr(0), _radcorr(0), _tracorr(0) {}
+   EFieldCorr(Double_t zcorr, Double_t radcorr, Double_t tracorr) : _zcorr(zcorr), _radcorr(radcorr), _tracorr(tracorr)
+   {
+   }
+   inline void SetZCorr(Double_t zcorr) { _zcorr = zcorr; }
+   inline void SetRadCorr(Double_t radcorr) { _radcorr = radcorr; }
+   inline void SetTraCorr(Double_t tracorr) { _tracorr = tracorr; }
+   inline Double_t GetZCorr() { return _zcorr; }
+   inline Double_t GetRadCorr() { return _radcorr; }
+   inline Double_t GetTraCorr() { return _tracorr; }
+
+   Double_t _zcorr;
+   Double_t _radcorr;
+   Double_t _tracorr;
+};
+
+class EFieldMapRef {
+
+public:
+   EFieldMapRef();
+   EFieldMapRef(Int_t rad, Int_t zpos);
+   Int_t _rad{0};  //! radius integer value
+   Int_t _zpos{0}; //! z integer value
+};
+
+bool operator<(const EFieldMapRef &l, const EFieldMapRef &r);
+bool operator==(const EFieldMapRef &l, const EFieldMapRef &r);
+std::ostream &operator<<(std::ostream &os, const EFieldMapRef &t);
+
+namespace std {
+template <>
+struct hash<EFieldMapRef> {
+   inline size_t operator()(const EFieldMapRef &x) const { return x._rad + x._zpos * 10000; }
+};
+} // namespace std
+
+using eDistortionMap = std::unordered_map<EFieldMapRef, EFieldCorr>; // Global LUT
+using BufferMap = std::unordered_map<EFieldMapRef, Double_t>;        //! To parse individual LUT
```

Comment:
I would put these in the AtDistortionModel class so they don't pollute the global namespace.